perllib/FixMyStreet/Cobrand/FixMyStreet.pm

```diff
@@ -310,6 +310,19 @@ sub updates_disallowed {
     return $self->next::method(@_);
 }
 
+sub problem_state_processed {
+    my ($self, $comment) = @_;
+
+    my $state = $comment->problem_state || '';
+    my $code = $comment->get_extra_metadata('external_status_code') || '';
+
+    my ($cfg) = $self->per_body_config('extra_state_mapping', $comment->problem);
+
+    $state = $cfg->{$state}->{$code} || $state;
```

Comment:
This was crashing with `Modification of a read-only value attempted` if `$cfg->{$state}` was undef, so something like this is needed:

```suggestion
    $state = ( $cfg->{$state}->{$code} || $state ) if $cfg->{$state};
```